### DISCUSSION TITLE
adding ansiblels configs

### DISCUSSION THREAD

**Discussion by Sam-su2060** [LazyVim/LazyVim#1453] [General]
Hello,
How to make ansbilels auto detect .yml, .yaml files ?
and how to enable snippets from all sources including lsp source when writing the playbook?
and how to set up auto indent based on the correct format?

Thanks,

**Comment by abeldekat** [comment]
Hello @Sam-su2060,

You could try the [nvim-ansible](https://github.com/mfussenegger/nvim-ansible) plugin.
This [recipe](https://astronvim.com/Recipes/yaml_ansible) is also interesting.

**Reply by Sam-su2060** [reply]
Thanks @abeldekat for your reply.

I will try the plugin later on.  
I already added the following autocmd in autocmds.lua
`vim.cmd("autocmd BufNewFile,BufRead *.yaml,*.yml set filetype=yaml.ansible")`
the autocmd attach ansiblels correctly. However, there is no auto-cmp settings to get snippets from lsp, and auto indent based on lsp..

So i need to add a custom file under .config/nvim/lua/plugins/ansible-lsp.lua, I tried something like that but no luck:
```
local cmp = require('cmp')

cmp.setup({
  snippet = {
    expand = function(args)
      require('luasnip').lsp_expand(args.body)
    end,
  },
  sources = {
    { name = 'nvim_lsp' },
    { name = 'luasnip' },
    { name = 'buffer' },
    { name = 'path' },
   
  },
})
```

**Reply by abeldekat** [reply]
I am afraid I can't help you regarding auto-cmp settings to get snippets from lsp. 
I thought the `nvim-ansible` plugin, the `ansiblels` lsp, and `LazyVim`'s `yaml` extra would be sufficient.
